### ISSUE TITLE
warn_indent not recognized as magic comment

### ISSUE THREAD

**Issue by rkh** [rubocop/rubocop#15607]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Layout/EmptyLineAfterMagicComment

### Steps to reproduce the problem

Create `example.rb`:

```ruby
# frozen_string_literal: true
# shareable_constant_value: literal
# warn_indent: true

puts "hi"
```

Run `rubocop example.rb`

### Expected behavior

Output:

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

### Actual behavior

Output:

```
Inspecting 1 file
C

Offenses:

example.rb:3:1: C: [Correctable] Layout/EmptyLineAfterMagicComment: Expected at least 1 empty line after magic comments; found 0.
# warn_indent: true
^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.90.0 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 3.4, running on ruby 4.0.6) [arm64-darwin23]
  - rubocop-minitest 0.40.0
  - rubocop-rake 0.7.1
```
